test/compilationTests/gnosis/Utils/Math.sol

```diff
@@ -94,12 +94,12 @@ library Math {
         zpow = zpow * z / ONE;
         result += 0x9c7 * zpow / ONE;
         if (shift >= 0) {
-            if (result >> (256-shift) > 0)
+            if (result >> uint(256-shift) > 0)
                 return (2**256-1);
-            return result << shift;
+            return result << uint(shift);
         }
         else
-            return result >> (-shift);
+            return result >> uint(-shift);
```

Comment:
This code may not do what it's supposed to do.

I think `result >> 256 - shift` is supposed to do a right shift if `shift > 256`. With the new code if `shift > 256`, then `result >> uint(256 - shift)` would always be a left shift.

Reply:
I'll update it to do the same as before.

Reply:
@mijovic But is `result >> uint(256-shift)` allowed by the typechecker?

Reply:
It is not allowed, that's why I added `uint(256 - shift)`